ixmp/core.py

```diff
@@ -1017,10 +1018,23 @@ def years_active(self, node, tec, yr_vtg):
         """
         return to_pylist(self._jobj.getTecActYrs(node, tec, str(yr_vtg)))
 
+    def get_meta(self, name=None):
+        meta = np.array(self._jobj.getMeta().entrySet().toArray()[:])
+        return {x.getKey(): unwrap(x.getValue()) for x in meta}
+
+    def set_meta(self, name, value):
+        self._jobj.setMeta(name, value)
 
 # %% auxiliary functions for class Scenario
 
 
+def unwrap(value):
```

Comment:
prefer defining this in the single function it is needed in, it's such a special case, I don't think it will be needed elsewhere, and if it is, it should probably be made more generic (e.g., jtype_to_pytype)